### PR TITLE
Register the LanguageServerLogger only once

### PR DESCRIPTION
**Pull request by dibarbet**
Fixes bug where logs were duplicated when using the LSP version.  Logged in https://github.com/OmniSharp/omnisharp-vscode/issues/5429

image

### CODE REVIEW

File: src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public class LanguageServerHost : IDisposable
                 .WithInput(input)
                 .WithOutput(output)
                 // initializeParams from the client won't be arriving yet, configure with app loglevel
-                .ConfigureLogging(GetLogBuilderAction(configureLogging, application.LogLevel))
+                .ConfigureLogging(AddLanguageProtocolLogging(application.LogLevel))
```
Comment (dibarbet) [inline src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs:65]: previously there were two calls to `GetLogBuilderAction`, one here and one when building the composition (after initialize).  Both calls registered the lsp logger and so we got double logs.

I modified this to register the LSP logger initially (with default level) and then configure it on composition building (after initialize).

File: src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs [hunk 2]
```diff
@@ -178,11 +178,12 @@ private static LogLevel GetLogLevel(InitializeTrace initializeTrace)
             }
         }
 
-        private static Action<ILoggingBuilder> GetLogBuilderAction(Action<ILoggingBuilder> configureLogging, LogLevel loglevel) => builder =>
+        private static Action<ILoggingBuilder> AddLanguageProtocolLogging(LogLevel loglevel) => builder => builder.AddLanguageProtocolLogging().SetMinimumLevel(loglevel);
+
+        private static Action<ILoggingBuilder> ConfigureLogging(Action<ILoggingBuilder> configureLogging, LogLevel loglevel) => builder =>
         {
             configureLogging?.Invoke(builder);
             builder
```
Comment (333fred) [inline src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs:186]: Nit: consider joining these lines.

File: src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs [hunk 3]
```diff
@@ -178,11 +178,12 @@ private static LogLevel GetLogLevel(InitializeTrace initializeTrace)
             }
         }
 
-        private static Action<ILoggingBuilder> GetLogBuilderAction(Action<ILoggingBuilder> configureLogging, LogLevel loglevel) => builder =>
+        private static Action<ILoggingBuilder> AddLanguageProtocolLogging(LogLevel loglevel) => builder => builder.AddLanguageProtocolLogging().SetMinimumLevel(loglevel);
```
Comment (333fred) [inline src/OmniSharp.LanguageServerProtocol/LanguageServerHost.cs:181]: Nit: consider wrapping this line.

**Review by 333fred — APPROVED** [review]
LGTM (commit 1), with a couple of formatting nits.

**Pull request merged**